### PR TITLE
Normalization layers fix (fixes #384 and #426).

### PR DESCRIPTION
**Pull request by eaplatanios**
This PR fixes the shape related issues in the normalization layers (i.e., fixes #384 and #426). The role of `LayerNorm` is unclear to me currently as I believe it's not really the same as the layer norm in the Python API. In either case, this PR fixes shape issues in a principled way. We can deal with what `LayerNorm` is supposed to be later.

cc @rxwei @BradLarson @mikowals

### CODE REVIEW

**Comment by eaplatanios** [comment]
@BradLarson could you please verify if your use cases works fine with these changes?

File: Sources/TensorFlow/Layers/Normalization.swift [hunk 1]
```diff
@@ -71,16 +71,26 @@ public struct BatchNorm<Scalar: TensorFlowFloatingPoint>: Layer {
...
+        let positiveAxis = (input.rank + axis) % input.rank
+        var offset = self.offset
+        var scale = self.scale
+        if positiveAxis != input.rank - 1 {
+            var broadcastShape = TensorShape([Int](repeating: 1, count: input.rank))
+            broadcastShape[positiveAxis] = input.shape[positiveAxis]
+            offset = offset.reshaped(to: broadcastShape)
+            scale = scale.reshaped(to: broadcastShape)
+            runningMean.value = runningMean.value.reshaped(to: broadcastShape)
+            runningVariance.value = runningVariance.value.reshaped(to: broadcastShape)
```
Comment (mikowals) [inline Sources/TensorFlow/Layers/Normalization.swift]: Are these reshapes of `runningMean.value` and `runningVariance.value` necessary?  I think either have rank 0 or the same rank as input, so they already always broadcast correctly.

Comment (eaplatanios) [reply Sources/TensorFlow/Layers/Normalization.swift]: Good point. I removed these reshapes.

**Comment by eaplatanios** [comment]
Great, thanks @BradLarson!

**Comment by mikowals** [comment]
@BradLarson I have an test that is part of #429.  I will make it part of a separate pull request later today.

**Comment by BradLarson** [comment]
@mikowals - Excellent, thanks. Better test coverage overall is definitely something we're interested in.

**Pull request merged**